### ISSUE TITLE
Fonts messed up during incremental typesetting

### ISSUE THREAD

**Issue by ktraunmueller** [ktraunmueller/Compositor#528]
There's a problem in the font loading / caching code that causes font definitions getting overwritten during incremental typesetting.

**Steps to reproduce**

Open this document:
[FontCaching.tex.zip](https://github.com/ktraunmueller/Compositor/files/5144995/FontCaching.tex.zip)

Click on the _a_ in the line _"here's some inline math"_ and bring up the formula editor (ctrl+f):
<img width="1592" alt="Screenshot 2020-08-29 at 11 16 03" src="https://user-images.githubusercontent.com/650763/91633563-9aae7e00-e9e9-11ea-9b27-d3f1ece01fda.png">

In the formula editor, click on the "a" and add a subscript by selecting the subscript formula building block:
<img width="1592" alt="Screenshot 2020-08-29 at 11 16 14" src="https://user-images.githubusercontent.com/650763/91633577-c0d41e00-e9e9-11ea-807e-46a3d2dc7fb5.png">

Notice that the fonts are still correct at this point. Specifically, look at the "n" in the _x subscript n_ on the same line.

Now select the placeholder (`\Box`) and type "n":
<img width="1592" alt="Screenshot 2020-08-29 at 11 16 35" src="https://user-images.githubusercontent.com/650763/91633593-ff69d880-e9e9-11ea-971d-f447bf2a908e.png">

At this point, the font used in the _x subscript n_ has been overwritten by another (upright) font.

Here are two trace logs that show in detail what happens between the two runs:
[FontsCorrect.txt](https://github.com/ktraunmueller/Compositor/files/5145001/FontsCorrect.txt)
[FontsMessedUp.txt](https://github.com/ktraunmueller/Compositor/files/5145002/FontsMessedUp.txt)

Things start going wrong in the line
```
locating tfm/OMX/lmex/m/n/14.tfm...
```
in `FontsMessedUp.txt`: the base file name is missing.

Note: Although this example uses a math formula to reproduce the issue, the problem is of a general nature and was not introduced with the formula editor.

**Comment by ktraunmueller** [comment]
As an observation: in this example, the issue is somehow linked to the `\thanks` command and the asterisk that it creates.

With verbose logging on, here's what happens when things go awry:
```
2020/10/26 12:47:13:436  looking up font ec-lmr7_R_458752...
2020/10/26 12:47:13:436  found font ec-lmr7_R_458752 as 42

>> =======================================================
capturing macro call: 4551 \T1\textasteriskcentered  (@369)
macro definition is:
\T1\textasteriskcentered ->\UseTextSymbol {OMS}\textasteriskcentered 
----------------------------------------------------------
start capturing macro parameters:
<< =======================================================
LaTeX Font Info:    Font shape `OMS/lmr/m/n' in size <7> not available
(Font)              Font shape `OMS/lmsy/m/n' tried instead on input line 18.
2020/10/26 12:47:13:440  looking up font lmsy7_R_458752...
2020/10/26 12:47:13:440  found font lmsy7_R_458752 as 28
2020/10/26 12:47:13:440  looking up font OMX/lmex/m/n/14_R_-1000...
2020/10/26 12:47:13:440  locating tfm/OMX/lmex/m/n/14.tfm...
2020/10/26 12:47:13:440  could not find tfm/OMX/lmex/m/n/14.tfm in distro
2020/10/26 12:47:13:441  could not find tfm/OMX/lmex/m/n/14.tfm in document folder
2020/10/26 12:47:13:441  failed to locate tfm/OMX/lmex/m/n/14.tfm
2020/10/26 12:47:13:441  tfm/OMX/lmex/m/n/14.tfm not found
```

So the `\textasteriskcentered` command is what's triggering the search for the OMX/lmex/m/n/14 tfm file.

**Comment by ktraunmueller** [comment]
After some more log reading, I think the problem is related to font size substitution:

In the second run (Fonts_SubscriptPlaceholder.txt), font `lmex10` is loaded (and registered as `lmex10_R_-1000`) for the requested size of 14.4:
```
LaTeX Font Info:    External font `lmex10' loaded for size
(Font)              <14.4> on input line 18.
2020/10/26 13:33:22:846  locating tfm/lmex10.tfm...
2020/10/26 13:33:22:847  io-trace > fopen(lmex10.tfm, rb)
2020/10/26 13:33:22:847  io-trace-cache > cache hit: /Users/ktraunmueller/Library/Developer/Xcode/DerivedData/Compositor-algvalfbyknlpvfwesfctugovthq/Build/Products/Debug/Compositor.app/Contents/Resources/texmf/tfm/lm/lmex10.tfm
2020/10/26 13:33:22:847  > registering font lmex10_R_-1000 as number 29
```
while in the faulty run (Fonts_SubscriptN.txt), font lookup is looking for `lmex` in size 14:
```
2020/10/26 13:36:01:052  looking up font OMX/lmex/m/n/14_R_-1000...
```

**Comment by ktraunmueller** [comment]
Fixed in 1.15
